src/coreclr/vm/clsload.cpp

```diff
@@ -2801,7 +2801,7 @@ TypeHandle ClassLoader::PublishType(const TypeKey *pTypeKey, TypeHandle typeHnd)
         {
             MethodDesc * pMD = it.GetMethodDesc();
             CONSISTENCY_CHECK(pMD != NULL && pMD->GetMethodTable() == pMT);
-            if (!pMD->IsUnboxingStub() && !pMD->IsAsyncHelperMethod())
+            if (!pMD->IsUnboxingStub() && !pMD->IsAsync2VariantMethod())
```

Comment:
We use `IsAsync2VariantMethod` to special case Async2 variant methods that were introduced in addition to what was defined in metadata. This is for scenarios that only want to deal with "real/defined" methods. 

As an example, if `Await` makes here - it is not a special case, it is a real defined method.